src/engine/controls/keycontrol.cpp

```diff
@@ -36,8 +36,8 @@ KeyControl::KeyControl(const QString& group,
 
     // pitch_adjust is the distance to the linear pitch in semitones
     // set by the speed slider or to the locked key.
-    // pitch_adjust knob in semitones; 4.7 ct per midi step; allowOutOfBounds = true;
-    m_pPitchAdjust = new ControlPotmeter(ConfigKey(group, "pitch_adjust"), -3.0, 3.0, true);
+    // pitch_adjust knob in semitones; 4.7 ct per midi step; allowOutOfBounds = false;
+    m_pPitchAdjust = new ControlPotmeter(ConfigKey(group, "pitch_adjust"), -15.0, 15.0, false);
```

Comment:
We have the same issue for "pitch_adjust", however her it is more complicated, because it is driven by m_pitchRateInfo.pitchTweakRatio or by m_pitchRateInfo.tempoRatio. 
You need to clamp these values and abort the operations early that all other variables a still consistent. 
The relationship is described here: 
https://github.com/mixxxdj/mixxx/blob/eff27abd571ac3e1b273ad4aef6407eceee0cfee/src/engine/controls/keycontrol.cpp#L154

The alternative would be to clamp the values before putting them into Soundthouch or Rubberband. But that might introduce transport issues because Mixxx does no longer uses the values for transport calculations that are actually used.